lib/install.js

```diff
@@ -662,42 +661,45 @@ Installer.prototype.printInstalled = function (cb) {
   validate('F', arguments)
   log.silly('install', 'printInstalled')
   var self = this
+  var removed = 0
+  var added = 0
+  var updated = 0
+  var moved = 0
   this.differences.forEach(function (action) {
     var mutation = action[0]
     var child = action[1]
     var name = packageId(child)
     var where = path.relative(self.where, child.path)
     if (mutation === 'remove') {
-      output('- ' + name + ' ' + where)
+      ++removed
     } else if (mutation === 'move') {
-      var oldWhere = path.relative(self.where, child.fromPath)
-      output(name + ' ' + oldWhere + ' -> ' + where)
+      ++moved
+    } else if (mutation === 'add') {
+      ++added
+    } else if (mutation === 'update' || mutation === 'update-linked') {
+      ++updated
+    } else {
+      log.warn('xxx', mutation)
     }
   })
-  var addedOrMoved = this.differences.filter(function (action) {
-    var mutation = action[0]
-    var child = action[1]
-    return !child.failed && (mutation === 'add' || mutation === 'update')
-  }).map(function (action) {
-    var child = action[1]
-    return child.path
-  })
-  if (!addedOrMoved.length) return cb()
-  // TODO: remove the recalculateMetadata, should not be needed
-  recalculateMetadata(this.idealTree, log, iferr(cb, function (tree) {
-    // These options control both how installs happen AND how `ls` shows output.
-    // Something like `npm install --production` only installs production deps.
-    // By contrast `npm install --production foo` installs `foo` and the
-    // `production` option is ignored. But when it comes time for `ls` to show
-    // its output, it excludes the thing we just installed because that flag.
-    // The summary output we get should be unfiltered, showing everything
-    // installed, so we clear these options before calling `ls`.
-    npm.config.set('production', false)
-    npm.config.set('dev', false)
-    npm.config.set('only', '')
-    npm.config.set('also', '')
-    ls.fromTree(self.where, tree, addedOrMoved, false, andIgnoreErrors(cb))
-  }))
+  var report = 'npm '
+  var actions = []
+  var plural = Math.max(added , removed , updated , moved) > 1
+  if (added) actions.push('added ' + added)
+  if (removed) actions.push('removed ' + removed)
+  if (updated) actions.push('updated ' + updated)
+  if (moved) actions.push('moved ' + moved)
+  if (actions.length === 0) {
+    return cb()
+  } else if (actions.length === 1) {
+    report += actions[0]
+  } else {
+    var lastAction = actions.pop()
+    report += actions.join(', ') + ' and ' + lastAction
+  }
+  report += ' package' + (plural ? 's' : '') + '.'
```

Comment:
I don't like complete sentences in logs. They look weird and takes 6 extra characters.